sound/soc/soc-pcm.c

```diff
@@ -626,7 +643,7 @@ int snd_soc_runtime_calc_hw(struct snd_soc_pcm_runtime *rtd,
 	unsigned int cpu_chan_min = 0, cpu_chan_max = UINT_MAX;
 	int i;
 
-	soc_pcm_hw_init(hw);
+	soc_pcm_hw_init(hw, false);
```

Comment:
Again, not a huge fan of these boolean parameters. The calling side readability suffers. Maybe soc_pcm_init_preserve_config() that hides the "true" case?